File: python/word-count/word_count.py
import re

def word_count(string):
    string = re.sub('[_]', ' ', string)
    string = re.sub('[,]', ' ', string)
    string = re.sub('[\s]', ' ', string)
    string = re.sub("n't", 'nZt', string)
    string = re.sub("'", ' ', string)
    string = re.sub('"', ' ', string)
    string = re.sub('[\W]', ' ', string)
    string = re.sub('nZt',"n't", string)
    string = string.lower()
    dictionary = {}
    split = string.split()
    for word in split:
        dictionary[word] = split.count(word)
    return dictionary

File: ruby/word-count/word_count.rb
class Phrase

  def initialize phrase
    @phrase = phrase
  end

  def words
    @phrase.downcase
           .gsub(" '", ' ')
           .gsub("' ", ' ')
           .gsub("'", 'Z')
           .gsub(',',' ')
           .gsub(/\W/, ' ')
           .gsub('Z', "'")
           .split
  end

  def word_count
    hash = {}
    words.uniq.each do |word|
      hash[word] = words.count(word)
    end
    hash
  end

end

module BookKeeping
  VERSION = 1
end
